scheduler/src/cook/kubernetes/api.clj

```diff
@@ -183,13 +185,15 @@
 (defn handle-watch-updates
   "When a watch update occurs (for pods or nodes) update both the state atom as well as
   invoke the callbacks on the previous and new values for the key."
-  [state-atom ^Watch watch key-fn callbacks]
+  [state-atom ^Watch watch key-fn callbacks set-metric-counters-fn]
   (while (.hasNext watch)
     (let [watch-response (.next watch)
           item (.-object watch-response)
           type (.-type watch-response)]
       (if (some? item)
-        (process-watch-response! state-atom item type key-fn callbacks)
+        (do
+          (process-watch-response! state-atom item type key-fn callbacks)
+          (set-metric-counters-fn))
```

Comment:
Would it be possible to just use `metrics.gauges/defgauge` instead of this approach? Seems like these are a good fit for a gauge.

Reply:
Decided offline to keep this as a counter for now